### ISSUE TITLE
yaidom-0004 removeAllInterElementWhitespace doesn't work when some of the children are comments

### ISSUE THREAD

**Issue by mhogerheijde** [dvreeze/yaidom#5]
When using the following XML in a scala worksheet

``` scala
import java.io.ByteArrayInputStream

import eu.cdevreeze.yaidom.parse.DocumentParserUsingSax

val xml = """<?xml version="1.0" encoding="UTF-8"?>
<rootElem>
  <listing>
    <item />
    <item />
    <!-- some comment -->
  </listing>
</rootElem>
"""

val inputStream = new ByteArrayInputStream(xml.getBytes)
val parser = DocumentParserUsingSax.newInstance()
val requestDoc = parser.parse(inputStream)
val documentElementWithoutWhitespace = requestDoc.documentElement.removeAllInterElementWhitespace
```

The content of `elemsWithoutWhitespace` still contains text nodes with white-space only around the `item` elements. The white-space around `listing` _is_ removed correctly.

As shown by this textual representation:

before whitespace removal:

```
documentElementWithWhitespace: eu.cdevreeze.yaidom.simple.Elem = elem(
  qname = QName("rootElem"),
  children = Vector(
    text("""
  """),
    elem(
      qname = QName("listing"),
      children = Vector(
        text("""
    """),
        elem(
          qname = QName("item")
        ),
        text("""
    """),
        elem(
          qname = QName("item")
        ),
        text("""
    """),
        comment(""" some comment """),
        text("""
  """)
      )
    ),
    text("""
""")
  )
)
```

after (incorrect) white-space removal

```
documentElementWithoutWhitespace: eu.cdevreeze.yaidom.simple.Elem = elem(
  qname = QName("rootElem"),
  children = Vector(
    elem(
      qname = QName("listing"),
      children = Vector(
        text("""
    """),
        elem(
          qname = QName("item")
        ),
        text("""
    """),
        elem(
          qname = QName("item")
        ),
        text("""
    """),
        comment(""" some comment """),
        text("""
  """)
      )
    )
  )
)
```

running the same code, without the comment results in the expected output

``` scala
import java.io.ByteArrayInputStream

import eu.cdevreeze.yaidom.parse.DocumentParserUsingSax

val xml = """<?xml version="1.0" encoding="UTF-8"?>
<rootElem>
  <listing>
    <item />
    <item />
  </listing>
</rootElem>
"""

val inputStream = new ByteArrayInputStream(xml.getBytes)
val parser = DocumentParserUsingSax.newInstance()
val requestDoc = parser.parse(inputStream)
val documentElementWithoutWhitespace = requestDoc.documentElement.removeAllInterElementWhitespace
```

results in what would be expected in the first example

```
documentElementWithoutWhitespace: eu.cdevreeze.yaidom.simple.Elem = elem(
  qname = QName("rootElem"),
  children = Vector(
    elem(
      qname = QName("listing"),
      children = Vector(
        elem(
          qname = QName("item")
        ),
        elem(
          qname = QName("item")
        )
      )
    )
  )
)
```

**Comment by dvreeze** [comment]
Sorry for suggesting the wrong title. Thanks for the clear description of the issue.

**Comment by dvreeze** [comment]
Fixed, and the fix will be included in yaidom 1.6.0.
